austation/code/game/objects/items/bluespace_jar.dm

```diff
@@ -221,38 +237,62 @@
+		var/mob/M = hit_atom
+		var/R = reagents.log_list()
+		hit_atom.visible_message("<span class='danger'>[M] has been splashed with something!</span>", \
+						"<span class='userdanger'>[M] has been splashed with something!</span>")
+		var/turf/TT = get_turf(hit_atom)
+		if(thrownby)
+			log_combat(thrownby, M, "splashed", R)
+			message_admins("[ADMIN_LOOKUPFLW(thrownby)] splashed (thrown) [english_list(reagents.reagent_list)] on [TT] at [ADMIN_VERBOSEJMP(TT)].")
+		reagents.reaction(hit_atom, TOUCH)
+		reagents.clear_reagents()
+
 	playsound(src, "shatter", 70, 1)
 	qdel(src)
 
 /obj/item/bluespace_jar/add_occupant(mob/living/occupant) //update the gas supply as required, this acts like magical internals
 	. = ..()
 	if(!occupant_gas_supply)
 		occupant_gas_supply = new
-	if(isanimal(occupant))
-//		var/mob/living/simple_animal/animal = occupant
-//		occupant_gas_supply.temperature = animal.minbodytemp //simple animals only care about temperature when their turf isnt a location
-	else
-		if(ishuman(occupant)) //humans require resistance to cold/heat and living in no air while inside, and lose this when outside
-			ADD_TRAIT(occupant, TRAIT_RESISTCOLD, "bluespace_container_cold_resist")
-			ADD_TRAIT(occupant, TRAIT_RESISTHEAT, "bluespace_container_heat_resist")
-			ADD_TRAIT(occupant, TRAIT_NOBREATH, "bluespace_container_no_breath")
-			ADD_TRAIT(occupant, TRAIT_RESISTHIGHPRESSURE, "bluespace_container_resist_high_pressure")
-			ADD_TRAIT(occupant, TRAIT_RESISTLOWPRESSURE, "bluespace_container_resist_low_pressure")
+	if(ishuman(occupant)) //humans require resistance to cold/heat and living in no air while inside, and lose this when outside
+		START_PROCESSING(SSobj, src)
+		ADD_TRAIT(occupant, TRAIT_RESISTCOLD, "bluespace_container_cold_resist")
+		ADD_TRAIT(occupant, TRAIT_RESISTHEAT, "bluespace_container_heat_resist")
+		ADD_TRAIT(occupant, TRAIT_NOBREATH, "bluespace_container_no_breath")
+		ADD_TRAIT(occupant, TRAIT_RESISTHIGHPRESSURE, "bluespace_container_resist_high_pressure")
+		ADD_TRAIT(occupant, TRAIT_RESISTLOWPRESSURE, "bluespace_container_resist_low_pressure")
 
 /obj/item/bluespace_jar/remove_occupant(mob/living/occupant)
 	. = ..()
 	if(ishuman(occupant))
+		STOP_PROCESSING(SSobj, src)
 		REMOVE_TRAIT(occupant, TRAIT_RESISTCOLD, "bluespace_container_cold_resist")
 		REMOVE_TRAIT(occupant, TRAIT_RESISTHEAT, "bluespace_container_heat_resist")
 		REMOVE_TRAIT(occupant, TRAIT_NOBREATH, "bluespace_container_no_breath")
 		REMOVE_TRAIT(occupant, TRAIT_RESISTHIGHPRESSURE, "bluespace_container_resist_high_pressure")
 		REMOVE_TRAIT(occupant, TRAIT_RESISTLOWPRESSURE, "bluespace_container_resist_low_pressure")
 
+/obj/item/bluespace_jar/process()
+	if(!reagents)
+		return
+	for(var/mob/living/L in occupants)
+		if(!ishuman(L))
+			continue
+		if((reagents.total_volume >= sipping_level) || ((reagents.total_volume >= sipping_probably) && prob(50))) //sipp
+			reagents.reaction(L, INGEST) //consume
+			reagents.trans_to(L, transfer_rate)
```

Comment:
tying this to prob(50) won't slow it down that much and isn't very efficient. Try making a new proc that has a world.time delay. You could use something like the process_reagents() proc from the pool controller

Reply:
do you know that process() is called every 10 deciseconds?

Reply:
the pool fires it every 2 deciseconds even

Reply:
Yes, Process fires every deci second, not once every 20. The pool fires it every 5 deciseconds with its current var, which is more than twice as slow as the jar. Alternatively you could use prob(20) or something similar

Reply:
it does fire once every 20 deciseconds, check the subsystem for obj

Reply:
pool uses fastprocessing instead of objprocessing

Reply:
🤔